tools/travis-before-install.sh

```diff
@@ -36,12 +36,5 @@ python -V
 popd
 
 pip install --upgrade pip setuptools
```

Comment:
tools/travis-test.sh:145 seems to install `setuptools` a second time, unless I'm missing something

Reply:
@eric-wieser, yes, I've noticed that the travis-ci scripts don't follow DRY very well.  `wheel` and `setuptools` are installed here and there.  Also, `cython` is in `test_requirement.txt`, but in fact `cython` is a build requirement.  I'm looking into this, so consider this PR to be WIP.

Reply:
I removed the redundant calls to `pip install` in the latest update.